pkg/client/api.go

```diff
@@ -178,13 +206,24 @@ func (h *queryClient) Series(ctx context.Context, interval model.Interval, selec
 	if err != nil {
 		return nil, err
 	}
+	humanReadableQuery := "Error, unable to read query"
+	if values, ok := vals["query"]; ok && len(values) > 0 {
+		humanReadableQuery = values[0]
+	}
 
 	var seriesRes []Series
-	err = json.Unmarshal(res.Data, &seriesRes)
-	return seriesRes, err
+	if err := json.Unmarshal(res.Data, &seriesRes); err != nil {
+		return nil, &Error{
+			Type:       ErrBadData,
+			ErrorMsg:   fmt.Sprintf("failed to unmarshal JSON response: %v", err),
+			StatusCode: res.StatusCode,
```

Comment:
Let's still use `err.StatusCode` from above in `res, err := h.api.Do(ctx, h.verb, seriesURL, vals)`. As much as possible, we want the metric status code to reflect the metrics source i.e. prom/m3 response code. Even if that call responded with 200 OK, we'll know this is an error because we'll increment the failed query metrics counter.

Reply:
If `err != nil` after `res, err := h.api.Do(ctx, h.verb, seriesURL, vals)`, then we return `err`.
Thus it makes no sense to try and use `err.StatusCode` from that call since it is guaranteed to be nil.
But the status code would be 200 OK at this point, so I will update to that value